lib/app/app.rb

```diff
@@ -163,18 +162,18 @@ class ConnectorsWebApp < Sinatra::Base
 
   # XXX remove `oauth2` from the name
   post '/oauth2/exchange' do
-    connector = settings.connector_class.new
-
     json connector.access_token(body_params)
   end
 
   post '/oauth2/refresh' do
-    connector = settings.connector_class.new
-
     json connector.refresh(body_params)
   end
 
   def body_params
     @body_params ||= JSON.parse(request.body.read, symbolize_names: true).with_indifferent_access
   end
+
+  def connector
+    settings.connector_class.new
```

Comment:
We'll need to cache this in case it's referenced multiple times, like in `/status`.

```suggestion
    @_connector ||= settings.connector_class.new
```